src/main/java/games/strategy/triplea/ui/ProductionRepairPanel.java

```diff
@@ -131,20 +130,19 @@ private void initRules(final PlayerID player, final Collection<PlayerID> allowed
           Match.getMatches(data.getMap().getTerritories(), Matches.territoryHasUnitsThatMatch(myDamagedUnits));
       for (final RepairRule repairRule : player.getRepairFrontier()) {
         for (final Territory terr : terrsWithPotentiallyDamagedUnits) {
-          for (final Unit u : Match.getMatches(terr.getUnits().getUnits(), myDamagedUnits)) {
-            if (!repairRule.getResults().keySet().iterator().next().equals(u.getType())) {
+          for (final Unit unit : Match.getMatches(terr.getUnits().getUnits(), myDamagedUnits)) {
+            if (!repairRule.getResults().keySet().iterator().next().equals(unit.getType())) {
               continue;
             }
-            final TripleAUnit taUnit = (TripleAUnit) u;
-            final Rule rule = new Rule(repairRule, player, m_uiContext, u, terr);
-            // int initialQuantity = initialPurchase.getInt(repairRule);
+            final TripleAUnit taUnit = (TripleAUnit) unit;
+            final Rule rule = new Rule(repairRule, player, unit, terr);
             int initialQuantity = 0;
-            if (initialPurchase.get(u) != null) {
-              initialQuantity = initialPurchase.get(u).getInt(repairRule);
+            if (initialPurchase.get(unit) != null) {
+              initialQuantity = initialPurchase.get(unit).getInt(repairRule);
```

Comment:
Yeah, big time :+1: 
It's interesting looking at a code base 10+ years old at how API design has evolved.